src/AggregateFunctions/AggregateFunctionGroupSortedArrayData.h

```diff
@@ -0,0 +1,174 @@
+#pragma once
+
+#include <IO/ReadBuffer.h>
+#include <IO/ReadHelpers.h>
+#include <IO/VarInt.h>
+#include <IO/WriteBuffer.h>
+#include <IO/WriteHelpers.h>
+
+#define DEFAULT_THRESHOLD 10
+
+namespace DB
+{
+template <typename Storage>
+struct AggregateFunctionGroupSortedArrayDataBase
```

Comment:
This is just a base class for common code for both std::multimap and std::set operations.